tests/stub/TestClock.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2018 KeePassXC Team <team@keepassxc.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 or (at your option)
+ * version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEEPASSXC_TESTCLOCK_H
+#define KEEPASSXC_TESTCLOCK_H
+
+#include "core/Clock.h"
+
+#include <QDateTime>
+
+namespace Test
+{
+    class Clock : public ::Clock
```

Comment:
Why not just call this call ClockTest or TestClock? ::Clock is confusing.

Reply:
Since only `Test::Clock` references `::Clock` it seems easier to write `Test::Clock` (shows that the object should only be used in tests) instead of `TestClock` (may collide with a potential test for `Clock`) or `Test::TestClock` (redundant naming) in the tests.

Reply:
I don't like this naming. It's against all our conventions. Wrap it in an anonymous namespace if you are worried about name clashes, but ::Clock is not the way to go.